includes/Core/Admin/Pointers.php

```diff
@@ -130,17 +143,64 @@ private function print_pointer_script( $pointer ) {
 									action:  "dismiss-wp-pointer",
 								}
 							);
+						},
+						pointerClass: "%s",
+						buttons: function( event, container ) {
+							return jQuery("<div></div>");
 						}
 					};
 
-					jQuery( "#%s" ).pointer( options ).pointer( "open" );
+					jQuery( "#%6$s" ).pointer( options ).pointer( "open" );
+
+					jQuery("body").on("click", ".googlesitekit-pointer-cta--dismiss", function() {
+						jQuery( "#%6$s" ).pointer("close");
+					});
+					jQuery("body").on("click", ".googlesitekit-pointer-cta", function(event) {
+						// We must prevent default link navigation in order to dismiss the pointer first
+						// to prevent it showing again.
+						event.preventDefault();
+						jQuery( "#%6$s" ).pointer("close");
+						window.location = jQuery(this).attr("href");
+						// TODO: add trackEvent in #11168
+					});
 				} );
 				',
-				esc_js( $pointer->get_title() ),
-				$content,
+				wp_kses(
+					$pointer->get_title(),
+					array(
+						'span'   => array( 'class' => array() ),
+						'button' => array(
+							'class' => array(),
+							'type'  => array(),
+						),
+					)
+				),
+				wp_kses(
+					$content,
+					array(
```

Comment:
Having these long arrays of rules for kses makes this hard to read since it's all inline in a formatted string call. Let's extract these to keep the `wp_kses` call here to preserve the late escape but keeping it to one line each.